internal/internal_decision_state_machine.go

```diff
@@ -371,14 +406,11 @@ func (d *commandStateMachineBase) cancel() {
 	case commandStateCompleted, commandStateCompletedAfterCancellationCommandSent:
 		// No op. This is legit. People could cancel context after timer/activity is done.
 	case commandStateCreated:
-		d.moveState(commandStateCompleted, eventCancel)
+		d.moveState(commandStateCanceledBeforeSent, eventCancel)
 	case commandStateCommandSent:
-		d.moveState(commandStateCanceledBeforeInitiated, eventCancel)
+		d.moveState(commandStateCancellationCommandSent, eventCancel)
 	case commandStateInitiated:
 		d.moveState(commandStateCanceledAfterInitiated, eventCancel)
-		// cancel doesn't add new command, therefore addCommand is not called.
-		// But *CancelRequested event is still being added to the history, therefore counter needs to be incremented.
-		d.helper.incrementNextCommandEventID()
```

Comment:
Will this break backwards compatibility?

Reply:
Per our discussion, there is one change in `internal_worker_test`: https://github.com/temporalio/sdk-go/pull/302/files#diff-2c7ff428b906672420092993fc82f49bcee4deff1806b693e2bad3c9eb59a13a where the ordering of activity test scheduled & cancel requested have swapped. Would that constitute a breaking change?

Reply:
There are no changes to that test now.